cmd/fyne/commands/bundle.go

```diff
@@ -12,25 +12,157 @@ import (
 	"strings"
 
 	"fyne.io/fyne/v2"
+	"github.com/urfave/cli/v2"
 )
 
 const fileHeader = "// auto-generated\n" + // to exclude this file in goreportcard (it has to be first)
 	"// Code generated by '$ fyne bundle'. DO NOT EDIT." // idiomatic mark, see https://golang.org/s/generatedcode
 
-// Declare conformity to Command interface
-var _ Command = (*bundler)(nil)
+// Bundle returns the vendor cli command
+func Bundle() *cli.Command {
+	b := &bundler{}
+
+	return &cli.Command{
+		Name:        "bundle",
+		Usage:       "Embeds static content into your go application.",
+		Description: "Each resource will have a generated filename unless specified.",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "name",
+				Usage:       "The variable name to assign the resource (file mode only).",
+				Destination: &b.name,
+			},
+			&cli.StringFlag{
+				Name:        "output",
+				Aliases:     []string{"o"},
+				Usage:       "Specify an output file instead of printing to standard output.",
+				Destination: &b.out,
+			},
+			&cli.StringFlag{
+				Name:        "package",
+				Aliases:     []string{"pkg"},
+				Usage:       "The package to output in headers (if not appending).",
+				Value:       "main",
+				Destination: &b.pkg,
+			},
+			&cli.StringFlag{
+				Name:        "prefix",
+				Usage:       "A prefix for variables (ignored if name is set).",
+				Value:       "resource",
+				Destination: &b.prefix,
+			},
+			&cli.BoolFlag{
+				Name:        "append",
+				Aliases:     []string{"a"},
+				Usage:       "Append an existing go file (don't output headers).",
+				Destination: &b.noheader,
+			},
+		},
+		Action: func(ctx *cli.Context) error {
+			if ctx.Args().Len() != 1 {
+				return errors.New("missing required file or directory parameter after flags")
+			}
+
+			outFile := os.Stdout
```

Comment:
Can this go into a separate function perhaps? The number of indentations in here is hard to read.
And harder to test.

Reply:
Can you help me also to see where this code moved from? The GitHub diff is not showing the source

Reply:
I have moved most of these out to separate functions now. It is taken from the Run() function.